File: Find Resultant Array After Removing Anagrams/Find Resultant Array After Removing Anagrams.py
# Solution for Find Resultant Array After Removing Anagrams in PY

class Solution:
    def checkAnagram(self, s1: str, s2: str) -> bool:
        arr = [0] * 26
        for ch in s1:
            arr[ord(ch) - ord('a')] += 1
        
        for ch in s2:
            arr[ord(ch) - ord('a')] -= 1
        
        for i in range(26):
            if arr[i] != 0:
                return False
        return True

    def removeAnagrams(self, words: List[str]) -> List[str]:
        result = [words[0]]
        for i in range(1, len(words)):
            if self.checkAnagram(result[-1], words[i]) == False:
                result.append(words[i])
        return result

File: Find Resultant Array After Removing Anagrams/Find Resultant Array After Removing Anagrams.java
// Solution for Find Resultant Array After Removing Anagrams in JAVA

class Solution {

    private boolean checkAnagram(String s1, String s2) {
        int[] arr = new int[26];
        for (char ch: s1.toCharArray()) {
            arr[ch - 'a']++;
        }
        for (char ch: s2.toCharArray()) {
            arr[ch - 'a']--;
        }
        for (int i = 0; i < 26; i++) {
            if (arr[i] != 0) {
                return false;
            }
        }
        return true;
    }

    public List<String> removeAnagrams(String[] words) {
        List<String> result = new ArrayList<>();
        result.add(words[0]);
        for (int i = 1; i < words.length; i++) {
            if (!checkAnagram(result.get(result.size() - 1), words[i])) {
                result.add(words[i]);
            }
        }
        return result;
    }
}
